public/app/plugins/datasource/elasticsearch/components/QueryEditor/MetricAggregationsEditor/SettingsEditor/MovingAverageSettingsEditor.tsx

```diff
@@ -1,159 +1,98 @@
-import { Input, InlineField, Select, Switch } from '@grafana/ui';
+import { Input, InlineField, Select, InlineSwitch } from '@grafana/ui';
 import React, { FunctionComponent } from 'react';
 import { useDispatch } from '../../../../hooks/useStatelessReducer';
 import { movingAvgModelOptions } from '../../../../query_def';
 import { isEWMAMovingAverage, isHoltMovingAverage, isHoltWintersMovingAverage, MovingAverage } from '../aggregations';
 import { changeMetricSetting } from '../state/actions';
+import { SettingField } from './SettingField';
 
 interface Props {
   metric: MovingAverage;
 }
 
 // The way we handle changes for those settings is not ideal compared to the other components in the editor
+// FIXME: using `changeMetricSetting` will cause an error when switching from models that have different options
+// as they might be incompatible. We should clear all other options on model change.
 export const MovingAverageSettingsEditor: FunctionComponent<Props> = ({ metric }) => {
   const dispatch = useDispatch();
 
   return (
     <>
-      <InlineField label="Model">
+      <InlineField label="Model" labelWidth={16}>
         <Select
           onChange={(value) => dispatch(changeMetricSetting(metric, 'model', value.value!))}
           options={movingAvgModelOptions}
           value={metric.settings?.model}
         />
       </InlineField>
 
-      <InlineField label="Window">
-        <Input
-          onBlur={(e) => dispatch(changeMetricSetting(metric, 'window', parseInt(e.target.value!, 10)))}
-          defaultValue={metric.settings?.window}
-        />
-      </InlineField>
-
-      <InlineField label="Predict">
-        <Input
-          onBlur={(e) => dispatch(changeMetricSetting(metric, 'predict', parseInt(e.target.value!, 10)))}
-          defaultValue={metric.settings?.predict}
-        />
-      </InlineField>
+      <SettingField label="Window" settingName="window" metric={metric} placeholder="5" />
 
-      {isEWMAMovingAverage(metric) && (
-        <>
-          <InlineField label="Alpha">
-            <Input
-              onBlur={(e) => dispatch(changeMetricSetting(metric, 'alpha', parseInt(e.target.value!, 10)))}
-              defaultValue={metric.settings?.alpha}
-            />
-          </InlineField>
+      <SettingField label="Predict" settingName="predict" metric={metric} />
 
-          <InlineField label="Minimize">
-            <Switch
-              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
-                dispatch(changeMetricSetting(metric, 'minimize', e.target.checked))
-              }
-              checked={!!metric.settings?.minimize}
-            />
-          </InlineField>
-        </>
+      {(isEWMAMovingAverage(metric) || isHoltMovingAverage(metric) || isHoltWintersMovingAverage(metric)) && (
+        <InlineField label="Alpha" labelWidth={16}>
```

Comment:
Could we use `SettingField` here? (and for other fields that are using `InlineField`)

Reply:
I guess we can't coz it requires entire settings object 🤔 

And Now for Something Completely Different: `alpha` prop is moving from `metric` to `metric.settings` for EWMA. It means that whatever user had saved in alpha will be gone after this change. Is it something worth mentioning as a breaking change?

Reply:
> I guess we can't coz it requires entire settings object

yeah, that's eventually something I want to change as I don't really like it as it is